### PR TITLE
Prevent magic value of windows OS

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind cleanup
/sig node

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 1]
```diff
@@ -142,7 +146,7 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxConfig(pod *v1.Pod, attemp
 	}
 	podSandboxConfig.Linux = lc
 
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == os_windows {
```
Comment (saschagrunert) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:149]: We can use this one:
https://github.com/kubernetes/kubernetes/blob/487c44323922fa9a0d217ac3a70eb5943a121fbb/pkg/apis/core/types.go#L3115

Comment (kerthcet) [reply pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:149]: Thanks, updated.

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 2]
```diff
@@ -37,6 +38,10 @@ import (
 	netutils "k8s.io/utils/net"
 )
 
+const (
+	os_windows = string(core.Windows)
```
Comment (bart0sh) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:42]: Isn't core.Windows a string already?

Comment (kerthcet) [reply pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:42]: `core.Windows` is type of OSName, which is a Type definition of `string`

**Comment by bart0sh** [comment]
From the first look this change looked reasonable. However, considering overall readability, low probability that GOOS "windows" would ever be changed and consistency ("windows" used all over the k/k codebase as is), I'm not sure we need this.
@SergeyKanzhelev @saschagrunert WDYT?

**Comment by kerthcet** [comment]
I'll close this if everyone has interest with this PR, plz go ahead.
/close

**Comment by saschagrunert** [comment]
> From the first look this change looked reasonable. However, considering overall readability, low probability that GOOS "windows" would ever be changed and consistency ("windows" used all over the k/k codebase as is), I'm not sure we need this. @SergeyKanzhelev @saschagrunert WDYT?

Alright, sounds good to me :+1:

**Pull request closed without merging**